projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.native.tsx

```diff
@@ -0,0 +1,170 @@
+} from '../../../../../state/constants';
+/**
+ * Types
+ */
+import type { VideoControlProps } from '../../types';
+import type React from 'react';
+
+/**
+ * Sidebar Control component.
+ *
+ * @param {VideoControlProps} props - Component props.
+ * @returns {React.ReactElement}    Component template
+ */
+export default function PrivacyAndRatingPanel( {
+	attributes,
+	setAttributes,
+	privateEnabledForSite,
+}: VideoControlProps ): React.ReactElement {
+	const [ showSubSheet, setShowSubSheet ] = useState( false );
+	const navigation = useNavigation();
+
+	const goBack = useCallback( () => {
+		setShowSubSheet( false );
+		navigation.goBack();
+	}, [] );
+
+	const openSubSheet = useCallback( () => {
+		navigation.navigate( BottomSheet.SubSheet.screenName );
+		setShowSubSheet( true );
+	}, [] );
+
+	const { privacySetting, rating, allowDownload, displayEmbed } = attributes;
+
+	const privacyLabels = {
+		private: _x( 'Site Default (Private)', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+		public: _x( 'Site Default (Public)', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+	};
+
+	const defaultPrivacyLabel = privateEnabledForSite ? privacyLabels.private : privacyLabels.public;
+
+	return (
+		<BottomSheet.SubSheet
+			navigationButton={
+				<BottomSheet.Cell
+					label={ __( 'Privacy and Rating', 'jetpack-videopress-pkg' ) }
+					onPress={ openSubSheet }
+					leftAlign
+				>
+					<Icon icon={ chevronRight }></Icon>
+				</BottomSheet.Cell>
+			}
+			showSheet={ showSubSheet }
+		>
+			<>
+				<BottomSheet.NavBar>
+					<BottomSheet.NavBar.BackButton onPress={ goBack } />
+					<BottomSheet.NavBar.Heading>
+						{ __( 'Privacy and Rating', 'jetpack-videopress-pkg' ) }
+					</BottomSheet.NavBar.Heading>
+				</BottomSheet.NavBar>
+
+				<PanelBody initialOpen={ false }>
+					<SelectControl
+						label={ _x( 'Rating', 'The age rating for this video.', 'jetpack-videopress-pkg' ) }
+						value={ rating ?? '' }
+						options={ [
+							{
+								label: _x( 'G', 'Video rating for "General Audiences".', 'jetpack-videopress-pkg' ),
+								value: VIDEO_RATING_G,
+							},
+							{
+								label: _x(
+									'PG-13',
+									'Video rating for "Parental Guidance", unsuitable for children under 13.',
+									'jetpack-videopress-pkg'
+								),
+								value: VIDEO_RATING_PG_13,
+							},
+							{
+								label: _x(
+									'R',
+									'Video rating for "Restricted", not recommended for children under 17.',
+									'jetpack-videopress-pkg'
+								),
+								value: VIDEO_RATING_R_17,
+							},
+						] }
+						onChange={ value => {
+							setAttributes( { rating: value } );
+						} }
+					/>
+
+					<SelectControl
+						label={ __( 'Privacy', 'jetpack-videopress-pkg' ) }
+						onChange={ value => {
+							const attrsToUpdate: {
+								privacySetting?: number;
+								isPrivate?: boolean;
+							} = {};
+
+							// Anticipate the isPrivate attribute.
+							if ( value !== '2' ) {
```

Comment:
_Nitpick:_ It's hard to know what the value `2` means in this logic. I understand that we could replace it with `String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_SITE_DEFAULT ) )` as it produces the same value. In this regard, maybe we could define the option values as constants to make it easier to read like:
```patch
diff --git forkSrcPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.native.tsx forkDstPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.native.tsx
index 6f8b05e4cf86fff86198e71a22893798f56da709..e99ea3d50717bb372fed261c7bc11a67e09f9562 100644
--- forkSrcPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.native.tsx
+++ forkDstPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.native.tsx
@@ -55,7 +55,18 @@ export default function PrivacyAndRatingPanel( {
 		public: _x( 'Site Default (Public)', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
 	};
 
-	const defaultPrivacyLabel = privateEnabledForSite ? privacyLabels.private : privacyLabels.public;
+	const privacyOptionSiteDefault = {
+		value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_SITE_DEFAULT ) ),
+		label: privateEnabledForSite ? privacyLabels.private : privacyLabels.public,
+	};
+	const privacyOptionPublic = {
+		value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_PUBLIC ) ),
+		label: _x( 'Public', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+	};
+	const privacyOptionPrivate = {
+		value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_PRIVATE ) ),
+		label: _x( 'Private', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+	};
 
 	return (
 		<BottomSheet.SubSheet
@@ -118,8 +129,8 @@ export default function PrivacyAndRatingPanel( {
 							} = {};
 
 							// Anticipate the isPrivate attribute.
-							if ( value !== '2' ) {
-								attrsToUpdate.isPrivate = value === '1';
+							if ( value !== privacyOptionSiteDefault.value ) {
+								attrsToUpdate.isPrivate = value === privacyOptionPrivate.value;
 							} else {
 								attrsToUpdate.isPrivate = privateEnabledForSite;
 							}
@@ -128,20 +139,7 @@ export default function PrivacyAndRatingPanel( {
 							setAttributes( attrsToUpdate );
 						} }
 						value={ String( privacySetting ) }
-						options={ [
-							{
-								value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_SITE_DEFAULT ) ),
-								label: defaultPrivacyLabel,
-							},
-							{
-								value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_PUBLIC ) ),
-								label: _x( 'Public', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
-							},
-							{
-								value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_PRIVATE ) ),
-								label: _x( 'Private', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
-							},
-						] }
+						options={ [ privacyOptionSiteDefault, privacyOptionPublic, privacyOptionPrivate ] }
 					/>
 
 					<ToggleControl
```

WDYT?

Reply:
Thanks Carlos, I agree this reads a lot better, I've gone ahead to implement the suggestion in https://github.com/Automattic/jetpack/pull/29477/commits/711e01cd2ead9dc55d6c0ddd87092d84ac2cb34d.